File: wwwroot/js/SacDeneModel.js
// API Bağlantı Bilgileri
const apiUrl = "https://hairstyle-changer-pro.p.rapidapi.com/facebody/editing/hairstyle-pro";
const apiKey = "b833913418msh34293360c392990p168517jsnfa01e09de061";

// Saç Modeli Uygulama Fonksiyonu
async function getStyledImage(hairStyle, imageFile) {
    try {
        // FormData oluştur
        const formData = new FormData();
        formData.append("file", imageFile, "image.jpg");
        formData.append("hair_style", hairStyle);
        formData.append("task_type", "hairstyle");

        // API çağrısı
        const response = await fetch(apiUrl, {
            method: "POST",
            headers: {
                "x-rapidapi-key": apiKey,
                "x-rapidapi-host": "hairstyle-changer-pro.p.rapidapi.com",
            },
            body: formData,
        });

        // Yanıtı kontrol et
        if (response.ok) {
            const jsonResponse = await response.json();
            return jsonResponse; // Başarılı yanıtı döndür
        } else {
            console.error("API Hatası:", response.status, response.statusText);
            return null; // Hata durumunda
        }
    } catch (error) {
        console.error("Bir hata oluştu:", error);
        return null;
    }
}

// Kullanıcıdan fotoğraf alıp API çağrısı yapan örnek kullanım
document.getElementById("uploadButton").addEventListener("click", async () => {
    const fileInput = document.getElementById("uploadPhoto");
    const selectedHairStyle = document.getElementById("hairStyleSelect").value;

    if (!fileInput.files.length || !selectedHairStyle) {
        alert("Lütfen bir fotoğraf seçin ve saç modeli belirleyin!");
        return;
    }

    const imageFile = fileInput.files[0];
    const result = await getStyledImage(selectedHairStyle, imageFile);

    if (result && result.image_url) {
        // Sonucu ekrana göster
        const outputImage = document.getElementById("outputImage");
        outputImage.src = result.image_url;
        outputImage.alt = "Yeni Saç Modeliniz";
    } else {
        alert("Saç modeli uygulama başarısız oldu!");
    }
});

File: Models/SacDeneModel.cs
using System.Net.Http;
using System.Threading.Tasks;
using System.Text.Json;
using System.Collections.Generic;

public class SacDeneModel
{
    private const string ApiUrl = "https://hairstyle-changer-pro.p.rapidapi.com/facebody/editing/hairstyle-pro";
    private const string ApiKey = "b833913418msh34293360c392990p168517jsnfa01e09de061";

    public async Task<string> GetStyledImage(string hairStyle, byte[] imageData)
    {
        try
        {
            using var client = new HttpClient();
            client.DefaultRequestHeaders.Add("x-rapidapi-key", ApiKey);
            client.DefaultRequestHeaders.Add("x-rapidapi-host", "hairstyle-changer-pro.p.rapidapi.com");

            var content = new MultipartFormDataContent
        {
            { new ByteArrayContent(imageData), "file", "image.jpg" },
            { new StringContent(hairStyle), "hair_style" },
            { new StringContent("hairstyle"), "task_type" }
        };

            var response = await client.PostAsync(ApiUrl, content);
            var jsonResult = await response.Content.ReadAsStringAsync();

            Console.WriteLine($"API Yanýtý: {jsonResult}");

            if (response.IsSuccessStatusCode)
            {
                var result = JsonSerializer.Deserialize<Dictionary<string, string>>(jsonResult);
                if (result != null && result.ContainsKey("image_url"))
                {
                    return result["image_url"];
                }
                else
                {
                    Console.WriteLine("API'den dönen yanýt beklenen formatta deðil: " + jsonResult);
                }
            }
            else
            {
                Console.WriteLine($"API Hatasý: {response.StatusCode}, Yanýt: {jsonResult}");
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine("Hata oluþtu: " + ex.Message);
        }

        return null;
    }


}
